mslib/mswms/mpl_lsec.py

```diff
@@ -0,0 +1,278 @@
+        tick_index_step = max(1, len(self.lat_inds) // int(self.numlabels))
+        ax.set_xticks(self.lat_inds[::tick_index_step])
+        ax.set_xticklabels([f"{d[0]:2.1f}, {d[1]:2.1f}"
+                            for d in zip(self.lats[::tick_index_step],
+                                         self.lons[::tick_index_step])],
+                           rotation=25, fontsize=10, horizontalalignment='right')
+
+        # Add lines to highlight points if any are given.
+        ipoint = 0
+        for i, (lat, lon) in enumerate(zip(self.lats, self.lons)):
+            if (ipoint < len(self.highlight) and
+                np.hypot(lat - self.highlight[ipoint][0],
+                         lon - self.highlight[ipoint][1]) < 2E-10):
+                ax.axvline(i, color='k', linewidth=2, linestyle='--', alpha=0.5)
+                ipoint += 1
+
+        # Set axis limits and draw grid for major ticks.
+        ax.set_xlim(self.lat_inds[0], self.lat_inds[-1])
+        ax.grid(b=True)
+
+    def _plot_style(self, color):
+        ax = self.ax
+
+        numpoints = len(self.lats)
+        ax.set_ylabel(self.unit)
+        ax.plot(range(numpoints), self.y_values, color.replace("0x", "#"))
+        self._latlon_setup()
+
+    def plot_lsection(self, data, lats, lons, valid_time, init_time,
+                      resolution=(-1, -1), style=None, show=False,
+                      highlight=None, noframe=False, figsize=(960, 480),
+                      numlabels=10, orography_color='k', transparent=False, color="0x00AAFF",
+                      return_format="image/png"):
+        """
+        """
+        # Check if required data is available.
+        self.data_units = self.driver.data_units.copy()
+        for datatype, dataitem, dataunit in self.required_datafields:
+            if dataitem not in data:
+                raise KeyError(f"required data field '{dataitem}' not found")
+            origunit = self.driver.data_units[dataitem]
+            if dataunit is not None:
+                data[dataitem] = convert_to(data[dataitem], origunit, dataunit)
+                self.data_units[dataitem] = dataunit
+            else:
+                logging.debug("Please add units to plot variables")
+
+        # Copy parameters to properties.
+        self.data = data
+        self.lats = lats
+        self.lat_inds = np.arange(len(lats))
+        self.lons = lons
+        self.valid_time = valid_time
+        self.init_time = init_time
+        self.resolution = resolution
+        self.style = style
+        self.highlight = highlight
+        self.noframe = noframe
+        self.numlabels = numlabels
+        self.orography_color = orography_color
+
+        # Derive additional data fields and make the plot.
+        self._prepare_datafields()
+
+        # Code for producing a png image with Matplotlib.
+        # ===============================================
+        if return_format == "image/png":
```

Comment:
this can probably be removed?

Reply:
It could. But currently requesting an image is as fully functional as an xml.
Should I get rid of linear images altogether?

Reply:
The only way to access those plots is with our client as it is non-standard WMS. If our client cannot/will not access the images, I think it should be removed before merging.

Reply:
I removed the linear image ability